### PR TITLE
Fix swap data

### PR DESCRIPTION
**Pull request by l-vincent-l**
Add unique identifier for each dataset.

### CODE REVIEW

**Comment by kinnou02** [comment]
jenkins, test this please

File: source/kraken/data_manager.h [hunk 1]
```diff
@@ -39,15 +39,19 @@ www.navitia.io
...
 #include <memory>
 #include <iostream>
+#include <atomic>
 #include <boost/make_shared.hpp>
 #include <boost/optional.hpp>
 
 template<typename Data>
 class DataManager{
     boost::shared_ptr<const Data> current_data;
+    std::atomic_size_t data_identifier;
```
Comment (antoine-de) [inline source/kraken/data_manager.h:49]: init it there directly no ?

File: source/kraken/data_manager.h [hunk 2]
```diff
@@ -57,8 +61,9 @@ class DataManager{
         release_memory();
     }
     boost::shared_ptr<const Data> get_data() const { return current_data; }
-    boost::shared_ptr<Data> get_data_clone() const {
-        auto data = boost::make_shared<Data>();
+    boost::shared_ptr<Data> get_data_clone() {
+        ++ data_identifier;
```
Comment (kinnou02) [inline source/kraken/data_manager.h:65]: why don't do this in `set_data` ?

File: source/type/data.cpp [hunk 3]
```diff
@@ -68,12 +68,20 @@ Data::Data() :
...
 {
-    this->is_connected_to_rabbitmq = false;
-    this->loaded = false;
+        loaded = false;
+        is_connected_to_rabbitmq = false;
 }
 
+Data::Data(size_t data_identifier) :
+    data_identifier(data_identifier) {
+        Data();
```
Comment (antoine-de) [inline source/type/data.cpp]: there is a problem here no ? why call Data() ?

File: source/kraken/data_manager.h (hunk 2 above)
Comment (l-vincent-l) [inline source/kraken/data_manager.h:65]: Because Data is const in set_data

File: source/kraken/data_manager.h (hunk 1 above)
Comment (l-vincent-l) [inline source/kraken/data_manager.h:49]: nop

**Comment by kinnou02** [comment]
jenkins, test this please

**Comment by l-vincent-l** [comment]
jenkins, test this please

**Pull request merged**